app/uk/gov/hmrc/vatregisteredcompanies/repositories/VatRegisteredCompaniesRepository.scala

```diff
@@ -110,35 +130,75 @@ class   VatRegisteredCompaniesRepository @Inject()(
     Future.successful((()))
   }
 
-  private def deleteById(deletes: List[BSONValue]): Future[Unit] = {
+  private def deleteById(deletes: List[Wrapper]): Future[Unit] = {
     if(deletes.nonEmpty) {
       logger.info(s"Deleting ${deletes.length} old entries")
       val source = Source(deletes)
-      val sink = Flow[BSONValue]
+      val sink = Flow[Wrapper]
         .map(_id =>
-          collection.findAndRemove(Json.obj("_id" -> _id), None, None, writeConcern = WriteConcern.Default, None, None, Seq.empty).map {_.result[BSONValue]}
-        ).to(Sink.onComplete { _ =>
+          //collection.findAndRemove(Json.obj("_id" -> _id), None, None, writeConcern = WriteConcern.Default, None, None, Seq.empty).map {_.result[BSONValue]}
+          collection.findOneAndDelete(Filters.equal("_id", _id))
+            .map{_=>
+              logger.info(s"Releasing lock ${_id}")
+              ()
+            })
+        .to(Sink.onComplete { _ =>
         logger.info("End of old entries deletion stream")
       })
       source.to(sink).run()
     }
     Future.successful((()))
   }
 
-  private def findOld(n: Int): Future[List[BSONDocument]] = {
```

Comment:
Is this function missed to be implemented?

Reply:
commented coded deleted